### PR TITLE
Fix ending newline bug

### PR DESCRIPTION
**Pull request by cmlsharp**
`style50` will sometimes not print the trailing newline after the diff. This PR addresses that bug, #6.

### CODE REVIEW

File: style50/core.py [hunk 1]
```diff
@@ -110,6 +110,7 @@ def run_diff(self):
             # Display results.
             if results.diffs:
                 print(*self.diff(results.original, results.styled), sep="",  end="")
+                print()
```
Comment (dmalan) [inline style50/core.py]: Why not just remove `end=""`?

Comment (cmlsharp) [reply style50/core.py]: Sorry about that, I was mistaken about what the source of the issue was. The latest commit addresses it. 

The real issue is that the IDE doesn't save files with a newline at the end (though most other editors do). This is addressed in the latest commit.

File: style50/core.py [hunk 2]
```diff
@@ -192,6 +191,13 @@ def unified(old, new):
         """
         Returns a generator yielding a unified diff between `old` and `new`.
         """
+        # Add \n to end of file if it doesn't already have it
+        if old[-1] != "\n":
+            old += "\n"
+
+        if new[-1] != "\n":
+            new += "\n"
+
```
Comment (dmalan) [inline style50/core.py]: Just to be sure, any issues if student runs `style50` on a blank file?

**Review by dmalan — DISMISSED** [review]

**Review by dmalan — DISMISSED** [review]

**Comment by cmlsharp** [comment]
Sorry for the many commits, it seems that each fix brings up a strange new edge case. I can squash when I merge if necessary

**Comment by dmalan** [comment]
Do just test before pushing to `develop`!

**Comment by cmlsharp** [comment]
Will do, sorry about that

File: style50/core.py [hunk 3]
```diff
@@ -167,18 +167,28 @@ def _check(self, file):
...
             with open(file) as f:
                 code = f.read()
+
         except (OSError, IOError) as e:
             if e.errno == errno.ENOENT:
                 raise Error("file \"{}\" not found".format(file))
             else:
                 raise
-        except KeyError:
+        except (KeyError, IndexError):
```
Comment (dmalan) [inline style50/core.py:179]: What's the different between these in this context?

File: style50/core.py [hunk 4]
```diff
@@ -167,18 +167,28 @@ def _check(self, file):
...
             raise Error("unknown file type \"{}\", skipping...".format(file))
+
+        # Ensure file ends in a trailing newline for consistency
+        try:
+            if code[-1] != "\n":
```
Comment (dmalan) [inline style50/core.py:184]: Will `len(code)` necessarily be `>= 1`?

**Review by dmalan — APPROVED** [review]

File: style50/core.py (hunk 4 above)
Comment (cmlsharp) [reply style50/core.py:184]: Nope, that's why it is in a `try`

File: style50/core.py (hunk 3 above)
Comment (cmlsharp) [reply style50/core.py:179]: They `IndexError` is actually probably not strictly necessary. I was going for accounting for the case that `extension` was fewer than 1 characters, but it seems that `""[1:]` returns the empty string instead of raising an `IndexError` (which is what I'd have expected)

File: style50/core.py (hunk 4 above)
Comment (dmalan) [reply style50/core.py:184]: That's definitely idiomatic?

File: style50/core.py (hunk 3 above)
Comment (dmalan) [reply style50/core.py:179]: K let's remove if not possible.
